### PR TITLE
implement save post

### PR DESCRIPTION
**Pull request by clarisaleu**
- save post is working

### CODE REVIEW

**Review by jocelynshen — APPROVED** [review]

File: app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java [hunk 1]
```diff
@@ -57,24 +72,73 @@ public void onClick(View v) {
...
+                    if (!mIsPostSaved) {
+                        // Post not saved yet - save
+                        SavedPost savedPost = new SavedPost();
+                        savedPost.setPost(mPost);
+                        savedPost.setUser(ParseUser.getCurrentUser());
+                        savedPost.saveInBackground(new SaveCallback() {
+                            @Override
+                            public void done(ParseException e) {
+                                mIsPostSaved = true;
+                                Toast.makeText(PostDetailActivity.this, "Saved Post", Toast.LENGTH_SHORT).show();
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java:93]: "Saved post" should be a string resource

File: app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java [hunk 2]
```diff
@@ -57,24 +72,73 @@ public void onClick(View v) {
...
+                        query.findInBackground(new FindCallback<ParseObject>() {
+                            @Override
+                            public void done(List<ParseObject> objects, ParseException e) {
+                                if (e == null) {
+                                    ParseObject.deleteAllInBackground(objects, new DeleteCallback() {
+                                        @Override
+                                        public void done(ParseException e) {
+                                            mIsPostSaved = false;
+                                            mBookmark.setImageResource(R.drawable.ic_save_check);
+                                            Toast.makeText(PostDetailActivity.this, "Unsaved Post", Toast.LENGTH_SHORT).show();
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java:110]: "Unsaved Post" should be a String resource

File: app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java [hunk 3]
```diff
@@ -57,24 +72,73 @@ public void onClick(View v) {
...
-                savedPost.setPost(post);
-                savedPost.setUser(ParseUser.getCurrentUser());
-                savedPost.saveInBackground(new SaveCallback() {
-                    @Override
-                    public void done(ParseException e) {
-                        Toast.makeText(PostDetailActivity.this, "Saved Post", Toast.LENGTH_LONG).show();
-                        // TODO: set tint
+            public void done(List<ParseObject> objects, ParseException e) {
+                if (e == null) {
+                    if (objects.size() > 0) {
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java:132]: If you're just checking if it's empty or not the preferred method is

if (!objects.isEmpty()) {

File: app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java [hunk 4]
```diff
@@ -86,9 +150,9 @@ public void done(ParseException e) {
     public void shareIntent(View view) {
         Intent shareIntent = new Intent();
         shareIntent.setAction(Intent.ACTION_SEND);
-        shareIntent.putExtra(Intent.EXTRA_TEXT, post.getTitle() + "\n" + post.getDescription());
-        if (post.getImage() != null && post.getImage().getUrl().length() > 0){
-            Uri imageUri = Uri.parse(post.getImage().getUrl());
+        shareIntent.putExtra(Intent.EXTRA_TEXT, mPost.getTitle() + "\n" + mPost.getDescription());
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java:153]: String.format

File: app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java [hunk 5]
```diff
@@ -57,24 +72,73 @@ public void onClick(View v) {
...
+                    if (objects.size() > 0) {
+                        mIsPostSaved = true;
+                        mBookmark.setImageResource(R.drawable.ic_save_check_filled);
+                    } else {
+                        mIsPostSaved = false;
+                        mBookmark.setImageResource(R.drawable.ic_save_check);
                     }
-                });
+                } else {
+                    Toast.makeText(PostDetailActivity.this, "Error querying saved posts", Toast.LENGTH_SHORT).show();
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java:140]: Toast messages should be in String resources

File: app/src/main/java/com/example/footprnt/Profile/Adapters/SavedPostsAdapter.java [hunk 6]
```diff
@@ -111,22 +118,74 @@ public void onResourceReady(@NonNull Bitmap resource, @Nullable Transition<? sup
...
+            public boolean onLongClick(View v) {
+                int newPosition = holder.getAdapterPosition();
+                SavedPost savedPost1 = mPosts.get(newPosition);
+                mPosts.remove(newPosition);
+                notifyItemRemoved(newPosition);
+                notifyItemChanged(newPosition, mPosts.size());
+                savedPost1.deleteInBackground(new DeleteCallback() {
+                    @Override
+                    public void done(ParseException e) {
+                        Toast.makeText(mContext, "Saved Post Deleted", Toast.LENGTH_SHORT).show();
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Profile/Adapters/SavedPostsAdapter.java:180]: Toast messages should be in String resources

**Pull request merged**